text/3700-ptr-assume-moved.md

```diff
@@ -0,0 +1,257 @@
+- Feature Name: `ptr_assume_moved`
+- Start Date: 2024-09-26
+- RFC PR: [rust-lang/rfcs#3700](https://github.com/rust-lang/rfcs/pull/3700)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add a helper for primitive pointer types to facilitate modifying the address of a pointer.  This
+mechanism is intended to enable the use of architecture features such as AArch64 Top-Byte Ignore
+(TBI) to facilitate use-cases such as high-bit pointer tagging.  An example application of this
+mechanism would be writing a tagging memory allocator.
+
+# Motivation
+[motivation]: #motivation
+
+The term "pointer tagging" could be used to mean either high-bit tagging or low-bit tagging.
+Architecture extensions such as AArch64 Top-Byte Ignore make the CPU disregard the top bits of a
+pointer when determining the memory address, leaving them free for other uses.
+
+This RFC is specifically concerned with creating those high-bit tagged pointers for systems which
+can make use of such architecture features. High-bit tagged pointers pose a somewhat tricky
+challenge for Rust, as the memory model still considers those high bits to be part of the address.
+Thus, from the memory model's perspective, changing those bits puts the pointer outside of its
+original allocation, despite it not being the case as far as the hardware & OS are concerned.  This
+makes loads and stores using the pointer Undefined Behaviour, despite the fact that if such loads
```

Comment:
```suggestion
makes loads and stores using the pointer undefined behaviour, despite the fact that if such loads
```

Please change this throughout.  We don't capitalize "undefined behavior" just as we don't capitalize "race condition".

Reply:
Yes, we've been cleaning this up incrementally.  Feel free to put in a PR there.